chronicler/events/core/git.py

```
@@ -143,3 +181,142 @@ def _process_action(self, source, time, event_uuid, action, file_data):
+
+    @staticmethod
+    def _get_sh_identity(git_user: str) -> dict:
+        """Extract identity information from a Git user string."""
+
+        identity = {}
+
+        fields = git_user.split("<")
+        name = fields[0]
+        name = name.strip()
+        email = None
+        if len(fields) > 1:
+            email = git_user.split("<")[1][:-1]
+        identity["username"] = None
+        identity["email"] = email
+        identity["name"] = name
+
+        return identity
+
+    @staticmethod
+    def _get_authors(authors_str: str) -> list[str]:
+        """Extract the authors from a multiauthor"""
+
+        authors = []
+
+        # Old style co-authored-by
+        m = AUTHOR_P2P_REGEX.match(authors_str)
+        if m:
+            authors = m.group("first_authors").split(",")
+            authors = [author.strip() for author in authors]
+            authors += [m.group("last_author")]
+
+        return list(set(authors))
+
+    def build_identity_event(
+        self,
+        identity: dict[str, Any],
+        role: str,
+        source_uri: str,
+        time: str,
+        event_type: str,
+        event_uuid: str
+    ):
+        """Build an identity event from the given identity information.
+
+        :param identity: dict with the identity information
+        :param role: role of the identity in the parent event
+        :param source_uri: data source of the event
+        :param time: time of the event
+        :param event_type: type of the identity event
+        :param event_uuid: UUID of the parent event
+
+        :returns: CloudEvent with the identity information
+        """
+        name = identity.get("name", None)
+        email = identity.get("email", None)
+        username = identity.get("username", None)
+
+        if not any([name, email, username]):
+            logger.warning(f"Identity for event '{event_uuid}' has no name, email or username. Skipping.")
+            return None
```

Probably is better to raise a defined exception and that the upper method captures and processes it. Then we avoid to have to handle `None` cases, that who know what that means.